packages/graphql/src/schema-model/annotation/Annotation.ts

```diff
@@ -17,6 +17,6 @@
  * limitations under the License.
  */
 
-import type { CypherAnnotation } from "./CypherAnnotation";
-
-export type Annotation = CypherAnnotation;
+export interface Annotation {
+    name: string;
+}
```

Comment:
@angrykoala, I remember that your intention was for this to be:

```ts
export type Annotation = AuthorizationAnnotation | CypherAnnotation
```

Was there a reason for this which might be scuppered by this?

Reply:
Good eye. Actually yes, this may be a problem, because typescript interfaces are stupid, this interface will be matched by anything that has a name property (regardless on wether or not the class implements that interface), that means that if we have a function such as
```ts
function doSomethingWithAnnotation(ann: Annotation)
```
The class `Node` for example, will be allowed to be passed into this, as it has a `name` property

The fix for this would either use an abstract class instead of an interface or the approach that @darrellwarde mentions, that explicitly says what things are annotations

Reply:
We changed a bit the implementation and moved back to the union type!